### PR TITLE
refactor: remove source activate

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Refactored conda envs within containers to be always on base. And changed rules to follow.

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by hassanfa** [comment]
I'm building and fixing py27 container. the reason is our base conda is py3, and conda doesn't like downgrading apparently.

File: BALSAMIC/snakemake_rules/annotation/rankscore.rule [hunk 1]
```diff
@@ -23,7 +22,6 @@ rule genmod_score_vardict:
     "and compress vcf using bcftools on {params.case_name}")
   shell:
     """
-source activate {params.conda};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/annotation/rankscore.rule:25]:
```suggestion
```

File: BALSAMIC/containers/varcall_py27/varcall_py27.sh [hunk 2]
```diff
@@ -1 +1 @@
-conda env create -n ${1} --file ${1}.yaml
+conda env update -n base --file ${1}.yaml
```
Comment (ashwini06) [inline BALSAMIC/containers/varcall_py27/varcall_py27.sh:1]:
```suggestion
conda env update -n base --file ${1}.yaml --prune?
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_normal.rule [hunk 3]
```diff
@@ -105,7 +100,6 @@ rule bcftools_filter_TNscope_umi_tumor_normal:
     "Filtering TNscope_umi tumor-normal annotated variants using bcftools on {params.case_name}"
   shell:
     """
-source activate {params.conda};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_normal.rule:103]:
```suggestion
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule [hunk 4]
```diff
@@ -67,7 +64,6 @@ rule bcftools_filter_tnhaplotyper_tumor_only:
     "Filtering tnhaplotyper tumor-only annotated variants using bcftools on {params.case_name}"
   shell:
     """
-source activate {params.conda};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule:67]:
```suggestion
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule [hunk 5]
```diff
@@ -101,7 +96,6 @@ rule bcftools_filter_TNscope_umi_tumor_only:
     "Filtering TNscope_umi tumor-only annotated variants using bcftools on {params.case_name}"
   shell:
     """
-source activate {params.conda};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule:99]:
```suggestion
```

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_only.rule [hunk 6]
```diff
@@ -32,7 +31,6 @@ rule manta_tumor_only:
         "index the compressed vcf file")
     shell:
       """
-source activate {params.conda};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_only.rule:34]:
```suggestion
```

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule [hunk 7]
```diff
@@ -88,7 +85,6 @@ rule delly_tumor_normal:
         "filter somatic variants and finally convert from bcf to compressed vcf file")
     shell:
         """
-source activate {params.conda};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule:88]:
```suggestion
```

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule [hunk 8]
```diff
@@ -36,7 +35,6 @@ rule manta_tumor_normal:
         "index the compressed vcf file")
     shell:
         """
-source activate {params.conda};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule:38]:
```suggestion
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule [hunk 9]
```diff
@@ -29,7 +28,6 @@ rule bcftools_filter_tnscope_tumor_only:
     "Filtering wgs tumor-only tnscope annotated variants using bcftools on {params.case_name}"
   shell:
     """
-source activate {params.conda};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_only.rule:31]:
```suggestion
```

**Review by ashwini06 — APPROVED** [review]
No more conda's 👍  
One minor+repetitive comment I have across all rules is an empty line(s) at the beginning of the shell script. Can you remove those empty lines 

`https://github.com/Clinical-Genomics/BALSAMIC/blob/b9dd944c96176e5385e81fba2eb7745cc8d65b85/BALSAMIC/snakemake_rules/variant_calling/mergetype_tumor.rule`

File: BALSAMIC/containers/align_qc/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+ARG WORK_DIR=project
+ARG CONTAINER_NAME
+
+# Copy all project files
+COPY . /${WORK_DIR}
+
+RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
+
+# Clean work environment
+RUN rm -rf /project && conda clean --all --yes
```
Comment (ivadym) [inline BALSAMIC/containers/align_qc/Dockerfile:24]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: BALSAMIC/containers/annotate/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+ARG WORK_DIR=project
+ARG CONTAINER_NAME
+
+# Copy all project files
+COPY . /${WORK_DIR}
+
+RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
+
+# Clean work environment
+RUN rm -rf /project && conda clean --all --yes
```
Comment (ivadym) [inline BALSAMIC/containers/annotate/Dockerfile:24]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: BALSAMIC/containers/balsamic/Dockerfile [hunk 12]
```diff
@@ -28,7 +28,7 @@ ARG CONTAINER_NAME
 COPY . /${WORK_DIR}
 
 RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
-RUN if [ "$CONTAINER_NAME" = "balsamic" ]; then cd /project && pip install --upgrade --no-cache-dir . ; fi
+RUN cd /${WORK_DIR} && pip install --upgrade --no-cache-dir .
 
 # Clean work environment
 RUN rm -rf /project && conda clean --all --yes
```
Comment (ivadym) [inline BALSAMIC/containers/balsamic/Dockerfile:34]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: BALSAMIC/containers/coverage_qc/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+ARG WORK_DIR=project
+ARG CONTAINER_NAME
+
+# Copy all project files
+COPY . /${WORK_DIR}
+
+RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
+
+# Clean work environment
+RUN rm -rf /project && conda clean --all --yes
```
Comment (ivadym) [inline BALSAMIC/containers/coverage_qc/Dockerfile:24]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: BALSAMIC/containers/varcall_cnvkit/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+ARG WORK_DIR=project
+ARG CONTAINER_NAME
+
+# Copy all project files
+COPY . /${WORK_DIR}
+
+RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
+
+# Clean work environment
+RUN rm -rf /project && conda clean --all --yes
```
Comment (ivadym) [inline BALSAMIC/containers/varcall_cnvkit/Dockerfile:24]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: BALSAMIC/containers/varcall_delly/Dockerfile [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+ARG WORK_DIR=project
+ARG CONTAINER_NAME
+
+# Copy all project files
+COPY . /${WORK_DIR}
+
+RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
+
+# Clean work environment
+RUN rm -rf /project && conda clean --all --yes
```
Comment (ivadym) [inline BALSAMIC/containers/varcall_delly/Dockerfile:24]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: BALSAMIC/containers/varcall_py27/Dockerfile [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+ARG WORK_DIR=project
+ARG CONTAINER_NAME
+
+# Copy all project files
+COPY . /${WORK_DIR}
+
+RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
+
+# Clean work environment
+RUN rm -rf /project && conda clean --all --yes
```
Comment (ivadym) [inline BALSAMIC/containers/varcall_py27/Dockerfile:24]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: BALSAMIC/containers/varcall_py36/Dockerfile [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+ARG WORK_DIR=project
+ARG CONTAINER_NAME
+
+# Copy all project files
+COPY . /${WORK_DIR}
+
+RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
+
+# Clean work environment
+RUN rm -rf /project && conda clean --all --yes
```
Comment (ivadym) [inline BALSAMIC/containers/varcall_py36/Dockerfile:24]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: BALSAMIC/containers/align_qc/Dockerfile (hunk 10 above)
Comment (ivadym) [inline BALSAMIC/containers/align_qc/Dockerfile:24]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: BALSAMIC/containers/annotate/Dockerfile (hunk 11 above)
Comment (ivadym) [inline BALSAMIC/containers/annotate/Dockerfile:24]:
```suggestion
RUN rm -rf /${WORK_DIR} && conda clean --all --yes
```

File: CHANGELOG.rst [hunk 18]
```diff
@@ -12,6 +12,7 @@ Changed:
 ^^^^^^^^
 
 * refactored main workflow with more readible organization #614
+* refactored conda envs within container to be on base #759
```
Comment (ashwini06) [inline CHANGELOG.rst:16]: * Update docker changes

File: BALSAMIC/containers/annotate/annotate.sh [hunk 19]
```diff
@@ -1,3 +1,2 @@
-conda env create -n ${1} --file ${1}.yaml
-source activate ${1}
+conda env update -n base --file ${1}.yaml
```
Comment (ashwini06) [inline BALSAMIC/containers/annotate/annotate.sh:1]:
```suggestion
conda env update -n base --file ${1}.yaml --prune
```

**Review by ivadym — APPROVED** [review]

**Pull request merged**